### DISCUSSION TITLE
Is there a technical reason the no extension works with css classes?

### DISCUSSION THREAD

**Discussion by sugoidesune** [ueberdosis/tiptap#1238] [Feature Requests]
I would like to use regular CSS classes instead of data-attributes.
Is  there a technical/architectural reason this isn't being used yet?
Or has it just not been done yet?
Thank you!

**Comment by philippkuehn** [comment]
you can use classes as well. I prefer classes for styling and data attributes to pass information.

**Reply by sugoidesune** [reply]
Just realized you have to create a new extension for every style-type: size, color, font etc. ( i need many many classes)
Was stuck with tradition HTML and wanted multiple classes on one. But It's tough for parsing i udnerstand.
```html
<span class="color-red font-arial">Text</span>
```

**Reply by Arik14** [reply]
Hello Mr. @sugoidesune , I just curious, do you finally create all those extensions?
Because I have closely related problem that require me to create many extensions too. The only different is I use style not class.

But there is problem. Whenever I put content from previous Tiptap version (in my app) it keep resetting the styling, like the font size.
Would you please help me? This is my issue https://github.com/ueberdosis/tiptap/discussions/1919
Thank you in advanced

**Comment by sugoidesune** [comment]
For others: This is what i extend the Highlight extension with to write classes
```js
 addAttributes() {
      return {
        value: {
          default: null,
          parseHTML: (element) => {
            return {
              value: element.getAttribute('class'),
            }
          },
          renderHTML: (attributes) => {
            if (!attributes.value) {
              return {}
            }

            return {
              class: attributes.value,
            }
          },
        },
      }
    },
```

**Reply by WangHaohui-org** [reply]
I have the same problem as you!plz help me!
